### PR TITLE
refactor(UserManager): Move methods to the manager

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
This moves the following methods to the manager:

- `createDM()`
- `deleteDM()`
- `fetchFlags()`

The `send()` method is also on the manager.

The `User` class isn't needed for these requests which in turn only made these possible via doing:

```JavaScript
<Client>.users.fetch("user_id").then(async user => {
  const dmChannel = await user.createDM();
  // ...
});
```

But now, only one API call will be made:

```JavaScript
<Client>.users.createDM("user_id").then(dmChannel => {
  // ...
});
```

A nice change from this is the `send()` method on the manager. To send a message to an uncached user, you would first have to fetch them (`UserManager#fetch()`) and then run `User#send()` which internally calls `User#createDM()`. That's 3 API requests, but on the manager, a DM is created first (`User#createDM()`) and then sent (`DMChannel#send()`). 1 less!

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

File: src/managers/UserManager.js [hunk 1]
```diff
@@ -32,26 +32,49 @@ class UserManager extends CachedManager {
    */
 
   /**
-   * Resolves a {@link UserResolvable} to a {@link User} object.
+   * The DM between the client's user and this user
```
Comment (SpaceEEC) [inline src/managers/UserManager.js:35]:
```suggestion
   * The DM between the client's user and a user
```

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
Just a small nit

File: src/managers/UserManager.js [hunk 2]
```diff
@@ -70,6 +93,50 @@ class UserManager extends CachedManager {
...
+    return (await this.fetch(user, options)).flags;
+  }
+
+  /**
+   * Sends a message to a user.
+   * @param {UserResolvable} user The UserResolvable to identify
+   * @param {string|MessagePayload|MessageOptions} options The options to provide
+   * @returns {Promise<Message>}
+   */
+
```
Comment (SpaceEEC) [inline src/managers/UserManager.js:113]:
```suggestion
```

**Review by SpaceEEC — DISMISSED** [review]
Otherwise lgtm

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**